### PR TITLE
Changelog Compile GitHub Action

### PR DESCRIPTION
**Pull request by MCterra10**
## About The Pull Request
Moves daily changelog compilation to a GitHub action.
If this is merged it will require the cron job that currently handles the compiles to be switched off or else things will get messy.

Changes tested downstream on AuStation and it works well.

## Why It's Good For The Game
This is GitHub

## Changelog
This is GitHub

### CODE REVIEW

**Comment by TheVekter** [comment]
Does this mean I won't get weird changelog commits/conflicts anymore?

**Comment by MCterra10** [comment]
I'm not sure what you mean.
The tgstation repo has a daily cron job that compiles and pushes the changelogs (deleting all the auto changelog files and packing them into the massive changelog file and updating the html)
This moves it to a GitHub action for convenience and to make changes simple.

**Comment by Cyberboss** [comment]
Can you test this locally on your own repo please?

**Comment by Cyberboss** [comment]
It's not a cron job IIRC. It's done by TGS when the servers are updated. Unless @optimumtact wants to correct me.

**Comment by MCterra10** [comment]
Yeah, it's a tgs4 script, but I thought it was running without tgs. My understanding was that tg still used tgs3 because v4 was unstable and buggy

**Comment by Cyberboss** [comment]
It's also a tgs3 script

**Comment by optimumtact** [comment]
there's a php webhook doing comment conversion, and tgs is doing the compile into the final product.

**Comment by MCterra10** [comment]
Ah well it seems I didn't understand how the script was setup, but it doesn't invalidate my PR. If it's tied to TGS though, it may have to be something that MSO deals with. If this goes through I'll work on moving changelog generation itself to an action

**Comment by Cyberboss** [comment]
The reason I bring it up is, if we accept this, it would have to be in addition to the TGS method. The servers compile the CL and push it when they update. Unless we want to stop the guarantee of having the most up-to-date changelogs at that moment, that method isn't going away because of this.

Honestly, a useful action would be one that could take over the entire process (Skip .yml generation and go straight to the html changelog on PR merge). That way, we could retire both the php webhook generator and the TGS scripts.

**Comment by MCterra10** [comment]
This is supposed to be a complete replacement for the current script, updated for the modern age. I made it for AuStation and PRd it here in case you are interested.
I'm not interested in changing this specific script at the moment as it serves its purpose exceedingly well downstream. I am interested in moving most of the infrastructure currently delt with by the webhook to actions (PR triage, CLs etc) in the future, however

**Comment by MCterra10** [comment]
Also austation's TGS compiles changelogs but just doesn't push, I would have thought it would be possible to have that here too

**Comment by optimumtact** [comment]
tgs is going to have to compile the changelogs irrespective because of test merges, but it doesn't need to push the results

**Comment by MCterra10** [comment]
the fix is in, this is ready to merge if you shut off tgs pushes

**Comment by optimumtact** [comment]
@MrStonedOne

**Comment by MrStonedOne** [comment]
tgs would only push the compiled changelogs if the change logs change from master after running compile. I'm not entirely sure why TGS's push would even need to be disabled for this pr to be merged.

They do not conflict with this pr and things would not in fact get messy. (in fact, all 6 servers currently handle pushing changelog updates when they automatically update).

Until somebody can give me an actual reason to turn off the tgs push of updated changelogs, i'm gonna leave them on since they don't actually conflict with this pr and I'm not entirely sure *why* maintainers are blocking this pr until those get disabled. It seems like that's all based off of an incorrect assumption.

**Comment by MrStonedOne** [comment]
@MCterra10 also: in the future could you be a bit more detailed in your pr descriptions. 

Details like when the github action runs should be included in the actual description and not be left for somebody to go code diving in the files tab to figure out.

In this case it runs every day at midnight gmt.

@tgstation/commit-access would we want that?

In fact, this is currently marked slower then the game servers that auto update every 6 hours, and there are 6 of them with their own offset since its 6 hours since the last manual update, meaning we should be getting a compile every 1 hour on avg.

**Comment by optimumtact** [comment]
I say we have both, so if tgs is borked for whatever reason we have a reasonable changelog backup.

The main thing I want running out of github itself is the conversion of comment PR's to changelog yaml files.

This is just a nice to have.

File: .github/workflows/compile_changelogs.yml
Comment (Cyberboss) [inline .github/workflows/compile_changelogs.yml]:
```suggestion
          git config --local user.email "tgstation-server@users.noreply.github.com"
```

File: .github/workflows/compile_changelogs.yml
Comment (Cyberboss) [inline .github/workflows/compile_changelogs.yml]: Use 
```suggestion
          git config --local user.name "tgstation-server"
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: .github/workflows/compile_changelogs.yml
Comment (optimumtact) [reply .github/workflows/compile_changelogs.yml]: disagree, this is not by that user

Comment (ninjanomnom) [reply .github/workflows/compile_changelogs.yml]: Yeah this is a github action, using that email is correct

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**
